homeassistant/components/tankerkoenig/__init__.py

```diff
@@ -1,174 +1,202 @@
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_API_KEY): cv.string,
+                    vol.Optional(
+                        CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                    ): cv.time_period,
+                    vol.Optional(CONF_FUEL_TYPES, default=FUEL_TYPES): vol.All(
+                        cv.ensure_list, [vol.In(FUEL_TYPES)]
+                    ),
+                    vol.Inclusive(
+                        CONF_LATITUDE,
+                        "coordinates",
+                        "Latitude and longitude must exist together",
+                    ): cv.latitude,
+                    vol.Inclusive(
+                        CONF_LONGITUDE,
+                        "coordinates",
+                        "Latitude and longitude must exist together",
+                    ): cv.longitude,
+                    vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS): vol.All(
+                        cv.positive_int, vol.Range(min=1)
+                    ),
+                    vol.Optional(CONF_STATIONS, default=[]): vol.All(
+                        cv.ensure_list, [cv.string]
+                    ),
+                    vol.Optional(CONF_SHOW_ON_MAP, default=True): cv.boolean,
+                }
+            )
+        },
+    ),
     extra=vol.ALLOW_EXTRA,
 )
 
+PLATFORMS = [Platform.SENSOR]
+
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set the tankerkoenig component up."""
     if DOMAIN not in config:
         return True
 
     conf = config[DOMAIN]
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_NAME: "Home",
+                CONF_API_KEY: conf[CONF_API_KEY],
+                CONF_FUEL_TYPES: conf[CONF_FUEL_TYPES],
+                CONF_LOCATION: {
+                    "latitude": conf.get(CONF_LATITUDE, hass.config.latitude),
+                    "longitude": conf.get(CONF_LONGITUDE, hass.config.longitude),
+                },
+                CONF_RADIUS: conf[CONF_RADIUS],
+                CONF_STATIONS: conf[CONF_STATIONS],
+                CONF_SHOW_ON_MAP: conf[CONF_SHOW_ON_MAP],
+            },
+        )
+    )
 
-    _LOGGER.debug("Setting up integration")
-
-    tankerkoenig = TankerkoenigData(hass, conf)
+    return True
 
-    latitude = conf.get(CONF_LATITUDE, hass.config.latitude)
-    longitude = conf.get(CONF_LONGITUDE, hass.config.longitude)
-    radius = conf[CONF_RADIUS]
-    additional_stations = conf[CONF_STATIONS]
 
-    setup_ok = await hass.async_add_executor_job(
-        tankerkoenig.setup, latitude, longitude, radius, additional_stations
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set a tankerkoenig configuration entry up."""
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][
+        entry.unique_id
```

Comment:
We should use the config entry entry_id instead to store the data in hass.data. This is standard.